src/cli/one_helper/onezone_helper.rb

```diff
@@ -577,7 +580,23 @@ def set_zone(zone_id, temporary_zone)
             puts "Endpoint changed to \"#{zone['TEMPLATE/ENDPOINT']}\" in " <<
                 "#{ENV['HOME']}/.one/one_endpoint"
         end
-        return 0
+
+        return 0 unless zone['TEMPLATE/ONEFLOW_ENDPOINT']
+
+        # Set ONEFLOW_ENDPOINT
+        if temporary_zone
+            puts "Type: export ONEFLOW_URL=#{zone['TEMPLATE/ONEFLOW_ENDPOINT']}"
+        else
+            File.open(ENV['HOME'] + '/.one/oneflow_endpoint', 'w') do |f|
+                f.puts zone['TEMPLATE/ONEFLOW_ENDPOINT']
+            end
+
+            puts 'OneFlow Endpoint changed to ' \
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against an allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html